dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/cache/DefaultCacheContext.java

```diff
@@ -0,0 +1,277 @@
+            .forceInMemory()
+            .withMaximumSize( orZeroInTestRun( 20000 ) )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createPeriodIdCache( Class<V> valueType )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "periodIdCache" )
+            .expireAfterWrite( 24, TimeUnit.HOURS )
+            .withInitialCapacity( 200 )
+            .forceInMemory()
+            .withMaximumSize( orZeroInTestRun( 10000 ) )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createUserAccountRecoverAttemptCache( Class<V> valueType, V defaultValue )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "userAccountRecoverAttempt" )
+            .expireAfterWrite( 15, TimeUnit.MINUTES )
+            .withDefaultValue( defaultValue )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createUserFailedLoginAttemptCache( Class<V> valueType, V defaultValue )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "userFailedLoginAttempt" )
+            .expireAfterWrite( 15, TimeUnit.MINUTES )
+            .withDefaultValue( defaultValue )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createProgramOwnerCache( Class<V> valueType )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "programOwner" )
+            .expireAfterWrite( 5, TimeUnit.MINUTES )
+            .withMaximumSize( orZeroInTestRun( 1000 ) )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createProgramTempOwnerCache( Class<V> valueType )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "programTempOwner" )
+            .expireAfterWrite( 30, TimeUnit.MINUTES )
+            .withMaximumSize( orZeroInTestRun( 4000 ) )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createUserIdCacheCache( Class<V> valueType )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "userIdCache" )
+            .expireAfterAccess( 1, TimeUnit.HOURS )
+            .withInitialCapacity( 200 )
+            .forceInMemory()
+            .withMaximumSize( orZeroInTestRun( 4000 ) )
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createCurrentUserGroupInfoCache( Class<V> valueType )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "currentUserGroupInfoCache" )
+            .expireAfterWrite( 1, TimeUnit.HOURS )
+            .forceInMemory()
+            .build();
+    }
+
+    @Override
+    public <V> Cache<V> createUserSettingCache( Class<V> valueType )
+    {
+        return cacheProvider.newCacheBuilder( valueType )
+            .forRegion( "userSetting" )
+            .expireAfterWrite( 12, TimeUnit.HOURS )
+            .withMaximumSize( orZeroInTestRun( 10000 ) )
```

Comment:
This might a good opportunity for applying `orZeroInTestRun` for all caches. That can of course also wait to avoid functional changes in this PR.

Reply:
I think we can risk it :D

Reply:
Yep, it would be really bad if tests started to fail then :-)